test/upgrade/docker-compose.yaml

```diff
@@ -0,0 +1,92 @@
+services:
+  mysql:
+    image: mysql:5.7
+    platform: linux/x86_64
+    command: mysqld --event-scheduler=ON
+    environment:
+      MYSQL_ROOT_PASSWORD: toor
+      MYSQL_DATABASE: fleet
+      MYSQL_USER: fleet
+      MYSQL_PASSWORD: fleet
+    ports:
+      - "3306"
+
+  redis:
+    image: redis:6
+
+  # reverse proxy and tls termination for fleet-a and fleet-b
+  fleet:
+    image: nginx
+    volumes:
+      # don't mount the config. These will be copied manually so that
+      # we can reload nginx without recreating containers and getting a new public port each time.
+      # - ./nginx/fleet-a.conf:/etc/nginx/conf.d/default.conf
+      - ./fleet.crt:/etc/nginx/fleet.crt
+      - ./fleet.key:/etc/nginx/fleet.key
+    ports:
+      - "443"
+
+  fleet-a:
+    image: fleetdm/fleet:${FLEET_VERSION_A:-latest}
+    environment:
+      FLEET_MYSQL_ADDRESS: mysql:3306
+      FLEET_MYSQL_DATABASE: fleet
+      FLEET_MYSQL_USERNAME: fleet
+      FLEET_MYSQL_PASSWORD: fleet
+      FLEET_REDIS_ADDRESS: redis:6379
+      FLEET_SERVER_ADDRESS: 0.0.0.0:8080
+      FLEET_SERVER_TLS: 'false'
+      FLEET_LOGGING_JSON: 'true'
+      FLEET_BETA_SOFTWARE_INVENTORY: 1
+      FLEET_LICENSE_KEY: ${FLEET_LICENSE_KEY}
+      FLEET_OSQUERY_LABEL_UPDATE_INTERVAL: 1m
+      FLEET_VULNERABILITIES_CURRENT_INSTANCE_CHECKS: "yes"
+      FLEET_VULNERABILITIES_DATABASES_PATH: /fleet/vulndb
+      FLEET_VULNERABILITIES_PERIODICITY: 5m
+      FLEET_LOGGING_DEBUG: 'true'
+      # This can be configured for testing purposes but otherwise uses the
+      # typical default of provided.
+      FLEET_OSQUERY_HOST_IDENTIFIER: ${FLEET_OSQUERY_HOST_IDENTIFIER:-provided}
+    ports:
+      - "8080"
+    depends_on:
+      - mysql
+      - redis
+
+  # Uses a different version than fleet-a
+  fleet-b:
+    image: fleetdm/fleet:${FLEET_VERSION_B:-latest}
+    platform: linux/x86_64
+    environment:
```

Comment:
just thinking out loud: can we use extension [fields](https://github.com/compose-spec/compose-spec/blob/master/spec.md#extension) ? this way the differences between both instances will be easier to spot (right now is only the version, right?)

Reply:
looks great! super clear what's the difference now. Thanks